x/qgb/orchestrator/parser.go

```diff
@@ -0,0 +1,67 @@
+package orchestrator
+
+import (
+	"fmt"
+
+	qgbtypes "github.com/celestiaorg/celestia-app/x/qgb/types"
+	sdkcodec "github.com/cosmos/cosmos-sdk/codec"
+	sdkcodectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdktypes "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/gogo/protobuf/proto"
+	coretypes "github.com/tendermint/tendermint/types"
+)
+
+var _ QGBParserI = QGBParser{}
+
+// FIXME remove this interface after writing tests and see if it is needed or not
+type QGBParserI interface {
+	ParseDataCommitmentConfirm(msg sdktypes.Msg) (qgbtypes.MsgDataCommitmentConfirm, error)
+	ParseValsetConfirm(msg sdktypes.Msg) (qgbtypes.MsgValsetConfirm, error)
+	ParseCoreTx(tx coretypes.Tx) (sdktx.Tx, error)
+	ParseSdkMsg(any *sdkcodectypes.Any) (sdktypes.Msg, error)
+}
+
+type QGBParser struct {
+	codec sdkcodec.Codec
+}
+
+func NewQGBParser(codec sdkcodec.Codec) *QGBParser {
+	return &QGBParser{codec: codec}
+}
+
+func (parser QGBParser) ParseValsetConfirm(msg sdktypes.Msg) (qgbtypes.MsgValsetConfirm, error) {
+	vs, ok := msg.(*qgbtypes.MsgValsetConfirm)
+	if !ok {
+		return qgbtypes.MsgValsetConfirm{},
+			fmt.Errorf("unable to cast sdk message to msg valset confirm")
+	}
+	return *vs, nil
+}
+
+func (parser QGBParser) ParseDataCommitmentConfirm(msg sdktypes.Msg) (qgbtypes.MsgDataCommitmentConfirm, error) {
+	dcc, ok := msg.(*qgbtypes.MsgDataCommitmentConfirm)
+	if !ok {
+		return qgbtypes.MsgDataCommitmentConfirm{},
+			fmt.Errorf("unable to cast sdk message to msg data commitment confirm")
+	}
+	return *dcc, nil
+}
+
+func (parser QGBParser) ParseCoreTx(tx coretypes.Tx) (sdktx.Tx, error) {
+	var sdkTx sdktx.Tx
+	err := proto.Unmarshal(tx, &sdkTx)
+	if err != nil {
+		return sdktx.Tx{}, fmt.Errorf("error while unmarshalling core transaction: %s", err)
+	}
+	return sdkTx, nil
+}
+
+func (parser QGBParser) ParseSdkMsg(any *sdkcodectypes.Any) (sdktypes.Msg, error) {
+	var stdMsg sdktypes.Msg
+	err := parser.codec.UnpackAny(any, &stdMsg)
+	if err != nil {
+		return nil, fmt.Errorf("error while unpacking message: %s", err)
+	}
+	return stdMsg, nil
+}
```

Comment:
[question] 
are we still using `Any`s anymore?

Reply:
yes, for attestations